File: roman-to-integer/roman_to_integer_test.py
import unittest
from roman_to_integer import Solution

class TestRomanToInt(unittest.TestCase):
  def setUp(self):
    self.testCases = [
      {
        's': 'III',
        'want': 3,
      },
      {
        's': 'IV',
        'want': 4,
      },
      {
        's': 'IX',
        'want': 9,
      },
      {
        's': 'LVIII',
        'want': 58,
      },
      {
        's': 'MCMXCIV',
        'want': 1994,
      },
    ]

  def test_want(self):
    s = Solution()
    for testCase in self.testCases:
      with self.subTest(testCase=testCase):
        self.assertEqual(s.romanToInt(testCase['s']), testCase['want'])

if __name__ == '__main__':
  unittest.main()

File: roman-to-integer/roman_to_integer_test.go
package romanToInt

import "testing"

func TestRomanToInt(t *testing.T) {
	testCases := []struct {
		s    string
		want int
	}{
		{s: "III", want: 3},
		{s: "IV", want: 4},
		{s: "IX", want: 9},
		{s: "LVIII", want: 58},
		{s: "MCMXCIV", want: 1994},
	}

	for idx, testCase := range testCases {
		if romanToInt(testCase.s) != testCase.want {
			t.Errorf("%d failed.\n", idx)
		}
	}
}
